### DISCUSSION TITLE
How can I rename a symbol across files (leveraging the LSP)?

### DISCUSSION THREAD

**Discussion by x10an14** [helix-editor/helix#4543] [Q&A] [answered]
Apologies in advance if there's some documentation that already answers this I haven't picked up.
If that's the case, please re-direct me there.

All I have managed is;
1. Check helix health:
    ```
    $ hx --health rust
    Configured language server: rust-analyzer
    Binary for language server: /nix/store/w3amgah9z90g4v5zvl037q9sq3l1igic-rust-analyzer-2022-10-10/bin/rust-analyzer
    Configured debug adapter: lldb-vscode
    Binary for debug adapter: /nix/store/ymjgfi7ml1n3gay2i8228w0sh3jr5424-lldb-14.0.6/bin/lldb-vscode
    Highlight queries: ✓
    Textobject queries: ✓
    Indent queries:  ✓
    ```
2. Open project with `hx .` for LSP purposes
3. Navigate to the item/symbol in question (function in my case)
4. Press SPACE+r (default) key-combo and re-name function
5. Watch `cargo build` fail for all files invoking this function, where it didn't get re-named

What am I doing wrong? =)

**Comment by David-Else** [comment, marked as answer]
The LSP hopefully did rename the functions over all the files, BUT it did not save those files for you.

If you press `space D` (open workspace diagnostic picker) or maybe G on the old version of Helix, it will probably show you all the errors in the files that have not been saved. You can go to each file using the picker and save them, Rust Analyzer needs the file to be saved to make the error go away.

Does that fix it? Happy Halloween :)

**Reply by sudormrfbin** [reply]
Or simply use `:write-all` to save all the files (helix opens files where a rename has been done).

**Reply by x10an14** [reply]
Thank you! =)
